src/imex.rs

```diff
@@ -643,13 +643,13 @@ async fn import_self_keys(context: &Context, path: &Path) -> Result<()> {
         } else {
             continue;
         };
-        let set_default = !name_f.contains("legacy");
         info!(
             context,
             "Considering key file: {}.",
             path_plus_name.display()
         );
 
+        let set_default = true;
         if let Err(err) = import_secret_key(context, &path_plus_name, set_default).await {
```

Comment:
`set_default` parameter of `import_secret_key()` and `set_self_key()` is always `true` now, could as well remove the parameter.

Maybe `KeyPairUse` can be removed as well.